### DISCUSSION TITLE
Assistance with custom endpoint for NER

### DISCUSSION THREAD

**Discussion by bholmessyapse** [explosion/spacy-llm#481] [Q&A] [answered]
Hi all - I'm trying to run an NER task using a mistral model, hosted within my organization, that I have access to through an endpoint like  https://FakeExample.endpoint.com/api/generate

I saw a feature was added in January that would allow for adding custom endpoints like this, and I attempted to add one - my config.cfg is given below

on running this, I get 

"Config validation error
llm.model -> endpoint   extra fields not permitted
"

I'm sure I'm doing something silly wrong, any advice is appreciated!

```
[nlp]
lang = "en"
pipeline = ["llm"]

[components]

[components.llm]
factory = "llm"

[components.llm.task]
@llm_tasks = "spacy.NER.v2"
labels = ["DISH", "INGREDIENT", "EQUIPMENT"]

[components.llm.model]
@llm_models = "spacy.Mistral.v1"
name = "Mistral-7B-v0.1"
endpoint = "https://FakeExample.endpoint.com/api/generate"

```

**Comment by bholmessyapse** [comment, marked as answer]
Figured out the solution - for anybody wondering, I had an ollama endpoint set up, and was able to query the LLM with a config like

```
[nlp]
lang = "en"
pipeline = ["llm"]
[components]
[components.llm]
factory = "llm"
[components.llm.task]
@llm_tasks = "spacy.NER.v3"
labels = ["label 1", "label 2", "label 3"]
[components.llm.model]
@llm_models = "langchain.Ollama.v1"
name =  "qwen2:7b"
config = {
    "base_url": "https://samplePlace.com",
    "temperature": 1.0
    }
```

**Reply by viradgupta** [reply]
Hi! I too have an open ai api provided by my organization. They have provided me with an enviromnent file which has below components in it:
OpenAI_API_Key = "XXXXX"
Base_URL = "https://gpt.company.com/XXXX"
API_Version = "2024-02-15-preview"
Model = "gpt-4-1106"

I am quite beginner level user of Python, just wanted your help on how do i integrate these items into the config file?
